src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -1625,7 +1640,16 @@ private static bool QueueUserWorkItemHelper(WaitCallback callBack, Object state,
                 try { }
                 finally
                 {
-                    QueueUserWorkItemCallback tpcallBack = new QueueUserWorkItemCallback(callBack, state, compressStack, ref stackMark);
+                    ExecutionContext context = compressStack && !ExecutionContext.IsFlowSuppressed() ?
+                                                ExecutionContext.Capture(ref stackMark, ExecutionContext.CaptureOptions.IgnoreSyncCtx | ExecutionContext.CaptureOptions.OptimizeDefaultCase) :
+                                                ExecutionContext.Default;
+
+                    QueueUserWorkItemCallback tpcallBack = (context == ExecutionContext.Default) ?
+                                                (state == null ? 
+                                                    new QueueUserWorkItemCallback(callBack) : // default or null context, no state 
+                                                    new QueueUserWorkItemCallbackWithState(callBack, state)) : // default or null context
```

Comment:
While it won't be something that ever occurs in CoreCLR, ```null``` actually needs to be handled by the context case so that ```ExecutionContext.Run``` is _not_ used with it.  ```ExecutionContext.Run(ExecutionContext.Default, ...)``` actually does some operations around context that should not be done in the null / flow suppressed case.